x/pairing/keeper/pairing.go

```diff
@@ -153,6 +153,42 @@ func (k Keeper) getPairingForClient(ctx sdk.Context, chainID string, clientAddre
 	return providers, vrfk, allowedCU, legacyStake, err
 }
 
+func (k Keeper) getProjectStrictestPolicy(ctx sdk.Context, project projectstypes.Project, chainID string) (uint64, uint64, string, uint64, error) {
+	plan, err := k.subscriptionKeeper.GetPlanFromSubscription(ctx, project.GetSubscription())
+	if err != nil {
+		return 0, 0, "", 0, err
+	}
+
+	planPolicy := plan.GetPlanPolicy()
+
+	err = project.VerifyProject(chainID, planPolicy)
+	if err != nil {
+		return 0, 0, "", 0, err
+	}
+
+	// geolocation is a bitmap. common denominator can be calculated with logical AND
+	geolocation := project.AdminPolicy.GeolocationProfile & project.SubscriptionPolicy.GeolocationProfile & planPolicy.GeolocationProfile
+
+	providersToPair := commontypes.FindMin([]uint64{
+		project.AdminPolicy.GetMaxProvidersToPair(),
+		project.SubscriptionPolicy.GetMaxProvidersToPair(),
+		planPolicy.GetMaxProvidersToPair(),
+	})
+
+	projectToPair := project.Index
+	allowedCU := commontypes.FindMin([]uint64{
+		project.AdminPolicy.GetEpochCuLimit(),
+		project.SubscriptionPolicy.GetEpochCuLimit(),
+		planPolicy.GetEpochCuLimit(),
+	})
+	err = project.VerifyCuUsage(planPolicy)
```

Comment:
move this verification to the relay payment instead of in the pairing. this is so the pairing wont change in the middle of the epoch after relay payments were done.